crates/miden-tx/src/host/mod.rs

```diff
@@ -930,6 +990,30 @@ where
     // HELPER FUNCTIONS
     // --------------------------------------------------------------------------------------------
 
+    /// Reads a double word (two [`Word`]s, 8 [`Felt`]s total) from the advice map.
+    ///
+    /// # Errors
+    /// Returns an error if the key is not present in the advice map or if the data is malformed
+    /// (not exactly 8 elements).
+    fn read_double_word_from_adv_map(
```

Comment:
nit: this probably could be a "free" helper function (i.e., it doesn't need to be a method on `TransactionBaseHost`).